src/upload/index.ts

```diff
@@ -0,0 +1,201 @@
+import {
+  HeadObjectCommand,
+  NotFound,
+  PutObjectCommand,
+  S3Client,
+} from "@aws-sdk/client-s3";
+import * as zlib from "zlib";
+import fs from "fs";
+
+import { LightBlockCache, lightBlockCache } from "@/cache";
+import { LightBlock } from "@/models/lightstreamer";
+import { logger } from "@/utils/logger";
+
+class UploadError extends Error {}
+
+export type BlockRange = {
+  start: number;
+  end: number;
+};
+
+export type BlockFile = {
+  file: string;
+  manifest: string;
+};
+
+export class LightBlockUpload {
+  private cache: LightBlockCache;
+  private s3Client: S3Client;
+  private chunkSizeMb: number;
+  private bucket: string;
+
+  constructor(cache: LightBlockCache) {
+    this.cache = cache;
+    if (!process.env["BUCKET_ENDPOINT"]) {
+      throw new UploadError("BUCKET_ENDPOINT not set");
+    }
+    if (!process.env["BUCKET_ACCESS_KEY_ID"]) {
+      throw new UploadError("BUCKET_ACCESS_KEY_ID not set");
+    }
+    if (!process.env["BUCKET_SECRET_ACCESS_KEY"]) {
+      throw new UploadError("BUCKET_SECRET_ACCESS_KEY not set");
+    }
+    if (!process.env["BUCKET_NAME"]) {
+      throw new UploadError("BUCKET_NAME not set");
+    }
+    if (!process.env["UPLOAD_CHUNK_SIZE_MB"]) {
+      throw new UploadError("UPLOAD_CHUNK_SIZE_MB not set");
+    }
```

Comment:
Nit:

```ts
for (const name of ["BUCKET_ENDPOINT", ...]) {
    if (!process.env[name]) { throw ... }
}
```

Reply:
The type checker complains about undefined assertions when I use the loop format (when constructing the `S3Client`, so I am going to leave as is.